src/apps/opencart/mod.rs

```diff
@@ -0,0 +1,46 @@
+use super::{App, State, Step};
+use futures::FutureExt;
+use thirtyfour::prelude::*;
+
+pub const APP: App = App {
+    test: &[
+        Step {
+            name: "login",
+            f: |st: &State| {
+                async {
+                    st.goto(st.url.join("index.php?route=account/login&language=en-gb")?.as_str()).await?;
+                    Ok(())
+                }
+                .boxed()
+            },
+            ..Step::default()
+        },
+        Step {
+            name: "admin-login",
+            f: |st: &State| {
+                async {
+                    st.goto(st.url.join("turnkey_admin")?.as_str()).await?;
```

Comment:
you don't need to join the path to `st.url` if you use `st.goto`, the function does it

Reply:
https://github.com/turnkeylinux/clicksnap/blob/master/src/apps/mod.rs#L83